crates/nu-std/std/testing.nu

```diff
@@ -1,5 +1,108 @@
 use log.nu
 
+# Here we store the list of annotations the test runner should concern itself with
+# Functions with no annotations or with annotations not on the list are rejected during module evaluation
+# Annotations starting with test- may be used multiple times throughout the module as the function names are stored in a list
+# Other annotations should only be used once within a module file
+# If you find yourself in need of multiple before- or after- functions it's a sign your test suite probably needs redesign
+def valid-annotations [] {
+    [
+        test,
+        test-skip,
+        before-each,
+        before-all,
+        after-each,
+        after-all
+
+    ]
+}
+
+
+# Returns a table containing the list of function names together with their annotations (comments above the declaration)
+#
+# The way this works is we first generate an ast for a file and then extract all tokens containing keyword def with internalcall shape
+# Then for each token we extract the next token (the actual function name) and the previous token (usually closing bracket of previous function)
+# We then open the file as raw text and extract the last line of substring starting from the span end of previous token up to span start of the def token which contains the annotation and surrounding whitespaces
+# We take the last line only in order to not pick up unrelated comments from the source code
+def get-annotated [
+    file: path
+] {
+    let raw_file = (open $file)
+
+    let ast = (
+        nu --ide-ast $file
```

Comment:
i think this should be `^$nu.current-exe --idae-ast $file` to generate the AST using the same `nu` binary as the one running the tests :+1:

Reply:
fixed

Reply:
thanks :sparkles: